core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/resolution/ResolutionModifier.java

```diff
@@ -72,26 +85,37 @@ public void setParameters(Map<String, String> params) throws ParametersException
       if (params.containsKey(HEIGHT_PARAM)) {
         height = NumberUtils.toInt(params.get(HEIGHT_PARAM));
         ParametersValidator
-            .checkRange(height, 1, MAX_SIZE, "Height should be greater than 0 and smaller than " + MAX_SIZE);
+            .checkRange(height, 1, MAX_SIZE,
+                "Height should be greater than 0 and smaller than " + MAX_SIZE);
+      } else {
+        samplingPeriod = Optional.ofNullable(params.get(SAMPLING_PERIOD_PARAM))
+            .map(NumberUtils::toInt)
```

Comment:
`NumberUtils.toInt(str)` returns `0` when conversion fails.
Consider using `NumberUtils.toInt(str, default)`
```suggestion
            .map(spp -> NumberUtils.toInt(spp, DEFAULT_SAMPLING_WAIT_PERIOD);
```

Reply:
Good catch, thank you.